### PR TITLE
change the position of the lock

### PR DESCRIPTION
**Pull request by pantianying**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #
image

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: registry/protocol/protocol.go [hunk 1]
```diff
@@ -338,10 +338,10 @@ func setProviderUrl(regURL *common.URL, providerURL *common.URL) {
 }
 
 func GetProtocol() protocol.Protocol {
```
Comment (flycash) [inline registry/protocol/protocol.go]: This method I had reverted to its origin version. You can merge master branch and there is not necessary to revert it again.

Comment (pantianying) [reply registry/protocol/protocol.go]: Yes, I know this, but I don't know how to merge your code into the development branch. This problem affects my test, so I added your recovered code to my branch

[the remark this answers is not in the document]
Comment (pantianying) [reply registry/protocol/protocol.go]: After the program runs, there are only read operations and no write operations. So I don't think there's a concurrency problem

**Comment by zouyx** [comment]
LGTM

**Pull request merged**